Source/FieldSolver/FiniteDifferenceSolver/FiniteDifferenceSolver.H

```diff
@@ -0,0 +1,56 @@
+#ifndef WARPX_FINITE_DIFFERENCE_SOLVER_H_
+#define WARPX_FINITE_DIFFERENCE_SOLVER_H_
+
+#include <AMReX_MultiFab.H>
+#include <WarpXAlgorithmSelection.H>
+#include<FiniteDifferenceAlgorithms/YeeAlgorithm.H>
+
+/**
+ * \brief Top-level class for the electromagnetic finite-difference solver
+ *
+ * TODO
+ */
+class FiniteDifferenceSolver
+{
+    public:
+
+        using VectorField = std::array< std::unique_ptr<amrex::MultiFab>, 3 >;
+
+        // Constructor
+        FiniteDifferenceSolver ( int fdtd_algo,
```

Comment:
But even better make this an enum type.
```suggestion
        FiniteDifferenceSolver ( int const fdtd_algo,
```